src/kt_variant.h

```diff
@@ -2,28 +2,356 @@
+        des->increment_position(encode_float(src_plane.d, des->get_cursor()));
+    }
+
+    static void to_kvariant_fromQUAT(SharedBuffer* des, const Variant& src) {
+        Quat src_quat{src.operator Quat()};
+        des->increment_position(encode_float(src_quat.x, des->get_cursor()));
+        des->increment_position(encode_float(src_quat.y, des->get_cursor()));
+        des->increment_position(encode_float(src_quat.z, des->get_cursor()));
+        des->increment_position(encode_float(src_quat.w, des->get_cursor()));
+    }
+
+    static void to_kvariant_fromAABB(SharedBuffer* des, const Variant& src) {
+        AABB src_aabb{src};
+        set_variant_type(des, Variant::Type::AABB);
+        append_vector3(des, src_aabb.position);
+        append_vector3(des, src_aabb.size);
+    }
+
+    static inline void append_basis(SharedBuffer* des, const Basis& data) {
+        append_vector3(des, data.get_axis(0));
+        append_vector3(des, data.get_axis(1));
+        append_vector3(des, data.get_axis(2));
+    }
+
+    static void to_kvariant_fromBASIS(SharedBuffer* des, const Variant& src) {
+        set_variant_type(des, Variant::Type::BASIS);
+        append_basis(des, src);
+    }
+
+    static void to_kvariant_fromTRANSFORM(SharedBuffer* des, const Variant& src) {
+        Transform src_transform{src.operator Transform()};
+        set_variant_type(des, Variant::Type::TRANSFORM);
+        append_basis(des, src_transform.basis);
+        append_vector3(des, src_transform.origin);
+    }
+
+    static void to_kvariant_fromARRAY(SharedBuffer* des, const Variant& src) {
+        set_variant_type(des, Variant::Type::ARRAY);
+        //TODO
+    }
+
+    static void to_kvariant_fromOBJECT(SharedBuffer* des, const Variant& src) {
+        Object* ptr{src};
+        set_variant_type(des, Variant::Type::OBJECT);
+
+        //TODO: Manage 32 bits systems.
+        des->increment_position(encode_uint64(reinterpret_cast<uintptr_t>(ptr), des->get_cursor()));
+
+        String class_name {ptr->get_class()};
+
+        if (!TypeManager::get_instance().JAVA_ENGINE_TYPES_CONSTRUCTORS.has(class_name)) {
+            class_name = ClassDB::get_parent_class(class_name);
+
+            while (class_name.empty()) {
+                if (!TypeManager::get_instance().JAVA_ENGINE_TYPES_CONSTRUCTORS.has(class_name)) {
+                    class_name = ClassDB::get_parent_class(class_name);
+                } else {
+                    break;
+                }
+            }
+        }
+
+        des->increment_position(encode_uint32(TypeManager::get_instance().JAVA_ENGINE_TYPES_CONSTRUCTORS[class_name], des->get_cursor()));
+        des->increment_position(encode_uint32(src.is_ref(), des->get_cursor()));
+        des->increment_position(encode_uint64(ptr->get_instance_id(), des->get_cursor()));
+    }
+
+    static void init_to_kt_methods(void (* to_kt_array[27 /* Variant::Type count */])(SharedBuffer*, const Variant&)) {
```

Comment:
This 27 appears some times in this file, maybe a const would make sense here for readability?

Reply:
Then also the comment would be obsolete